### ISSUE TITLE
Plugin loading fails in classic mode

### ISSUE THREAD

**Issue by rexx6** [springjools/deluge-autoremoveplus#29]
Environment : Owner@all4u:~# deluge -V
deluge 2.0.3
libtorrent: 1.1.13.0
Python: 3.8.2
OS: Linux Ubuntu 20.04 focal

Hi
When selecting a plugin, the following error message appears and the autoremoveplus interface is not displayed on the screen.
Thanks for helping

```
Owner@all4u:~# Unhandled error in Deferred:
Temporarily disabling observer LegacyLogObserverWrapper(<bound method TwistedLoggingObserver.emit of <deluge.log.TwistedLoggingObserver object at 0x7f6759b83310>>) due to exception: [Failure instance: Traceback: <class 'TypeError'>: _findCaller() takes from 1 to 2 positional arguments but 3 were given
/usr/lib/python3.8/json/encoder.py:253:iterencode
/usr/lib/python3/dist-packages/twisted/internet/defer.py:953:__del__
/usr/lib/python3/dist-packages/twisted/logger/_logger.py:270:critical
/usr/lib/python3/dist-packages/twisted/logger/_logger.py:144:emit
--- <exception caught here> ---
/usr/lib/python3/dist-packages/twisted/logger/_observer.py:131:__call__
/usr/lib/python3/dist-packages/twisted/logger/_legacy.py:93:__call__
/usr/lib/python3/dist-packages/deluge/log.py:208:emit
/usr/lib/python3/dist-packages/twisted/python/log.py:595:emit
/usr/lib/python3/dist-packages/twisted/logger/_legacy.py:154:publishToNewObserver
/usr/lib/python3/dist-packages/twisted/logger/_stdlib.py:115:__call__
/usr/lib/python3.8/logging/__init__.py:1500:log
/usr/lib/python3.8/logging/__init__.py:1565:_log
]
Traceback (most recent call last):
  File "/usr/lib/python3.8/json/encoder.py", line 253, in iterencode
    _iterencode = _make_iterencode(
  File "/usr/lib/python3/dist-packages/twisted/internet/defer.py", line 953, in __del__
    log.critical("Unhandled error in Deferred:",
  File "/usr/lib/python3/dist-packages/twisted/logger/_logger.py", line 270, in critical
    self.emit(LogLevel.critical, format, **kwargs)
  File "/usr/lib/python3/dist-packages/twisted/logger/_logger.py", line 144, in emit
    self.observer(event)
--- <exception caught here> ---
  File "/usr/lib/python3/dist-packages/twisted/logger/_observer.py", line 131, in __call__
    observer(event)
  File "/usr/lib/python3/dist-packages/twisted/logger/_legacy.py", line 93, in __call__
    self.legacyObserver(event)
  File "/usr/lib/python3/dist-packages/deluge/log.py", line 208, in emit
    PythonLoggingObserver.emit(self, event_dict)
  File "/usr/lib/python3/dist-packages/twisted/python/log.py", line 595, in emit
    _publishNew(self._newObserver, eventDict, textFromEventDict)
  File "/usr/lib/python3/dist-packages/twisted/logger/_legacy.py", line 154, in publishToNewObserver
    observer(eventDict)
  File "/usr/lib/python3/dist-packages/twisted/logger/_stdlib.py", line 115, in __call__
    self.logger.log(
  File "/usr/lib/python3.8/logging/__init__.py", line 1500, in log
    self._log(level, msg, args, **kwargs)
  File "/usr/lib/python3.8/logging/__init__.py", line 1565, in _log
    fn, lno, func, sinfo = self.findCaller(stack_info, stacklevel)
builtins.TypeError: _findCaller() takes from 1 to 2 positional arguments but 3 were given
```
![foo](https://user-images.githubusercontent.com/28583017/84095338-03650a80-aa3a-11ea-92de-688eb5fd51f0.png)

**Comment by springjools** [comment]
What version of twisted? Update to the latest version. Also I think you need deluge 2.0.4dev if you have python >= 3.8.

**Comment by rexx6** [comment]
Thanks for the answer.
I updated Twisted as below but the problem is the same.
```
Owner@all4u:~# pip3 show Twisted
Name: Twisted
Version: 20.3.0
Summary: An asynchronous networking framework written in Python
Home-page: https://twistedmatrix.com/
Author: Twisted Matrix Laboratories
Author-email: twisted-python@twistedmatrix.com
License: MIT
Location: /usr/local/lib/python3.8/dist-packages
Requires: constantly, incremental, attrs, Automat, hyperlink, PyHamcrest, zope.interface
Required-by: 
```
```
Owner@all4u:~# Unhandled error in Deferred:
09:35:58 [CRITICAL][twisted                           :154 ] Unhandled error in Deferred:
09:35:58 [CRITICAL][twisted                           :154 ] Unhandled error in Deferred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/deluge/ui/gtk3/pluginmanager.py", line 74, in _on_plugin_enabled_event
    self.enable_plugin(name)
  File "/usr/lib/python3/dist-packages/deluge/pluginmanagerbase.py", line 153, in enable_plugin
    log.error(
  File "/usr/local/lib/python3.8/dist-packages/twisted/internet/defer.py", line 1613, in unwindGenerator
    return _cancellableInlineCallbacks(gen)
  File "/usr/local/lib/python3.8/dist-packages/twisted/internet/defer.py", line 1529, in _cancellableInlineCallbacks
    _inlineCallbacks(None, g, status)
--- <exception caught here> ---
  File "/usr/local/lib/python3.8/dist-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "/usr/lib/python3/dist-packages/deluge/log.py", line 79, in error
    yield LoggingLoggerClass.error(self, msg, *args, **kwargs)
  File "/usr/lib/python3.8/logging/__init__.py", line 1463, in error
    self._log(ERROR, msg, args, **kwargs)
  File "/usr/lib/python3.8/logging/__init__.py", line 1565, in _log
    fn, lno, func, sinfo = self.findCaller(stack_info, stacklevel)
builtins.TypeError: findCaller() takes from 1 to 2 positional arguments but 3 were given

Temporarily disabling observer LegacyLogObserverWrapper(<bound method TwistedLoggingObserver.emit of <deluge.log.TwistedLoggingObserver object at 0x7f8f17b5f5e0>>) due to exception: [Failure instance: Traceback: <class 'TypeError'>: findCaller() takes from 1 to 2 positional arguments but 3 were given
/usr/lib/python3.8/json/encoder.py:253:iterencode
/usr/local/lib/python3.8/dist-packages/twisted/internet/defer.py:962:__del__
/usr/local/lib/python3.8/dist-packages/twisted/logger/_logger.py:190:failure
/usr/local/lib/python3.8/dist-packages/twisted/logger/_logger.py:144:emit
--- <exception caught here> ---
/usr/local/lib/python3.8/dist-packages/twisted/logger/_observer.py:131:__call__
/usr/local/lib/python3.8/dist-packages/twisted/logger/_legacy.py:93:__call__
/usr/lib/python3/dist-packages/deluge/log.py:204:emit
/usr/lib/python3.8/logging/__init__.py:1481:critical
/usr/lib/python3.8/logging/__init__.py:1565:_log
```

The ability to install deluge development version is not available, 
So I will give up.
Thank you.

**Comment by garretyoder** [comment]
I'm also having the same issue, however I downgraded to python 3.7 and still had the same issue.

**Comment by garretyoder** [comment]
I've also manually installed deluge 2.0.4 as mentioned and still receive the same error.

Edit: installing from git with the latest patch fixes this issue. The issue is not present in deluge==2.0.4.dev38.

**Comment by springjools** [comment]
If you downgrade to python 3.7 it should also fix the issue, but you have to make sure deluge uses that python.  Anyway, it's not an issue related to this plugin but to deluge itself. More info here:
https://dev.deluge-torrent.org/ticket/3327
